From c2af693a038c92503d829a13a57f22e070f2cfe4 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Mon, 10 Feb 2020 07:56:40 +0100
Subject: [PATCH 077/106] crypto: use $(PERL_PATH) instead of $(PERL)

Fixes the following building error with Android Q:

  PERLASM arch/x86/crypto/poly1305-x86_64-cryptogams.S
"perl" is not allowed to be used. See https://android.googlesource.com/platform/build/+/master/Changes.md#PATH_Tools for more information.
make[3]: *** [/home/utente/q-x86_kernel/kernel/arch/x86/crypto/Makefile:127: arch/x86/crypto/poly1305-x86_64-cryptogams.S] Error 1
---
 arch/x86/crypto/Makefile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/x86/crypto/Makefile b/arch/x86/crypto/Makefile
index a31de0c6ccde..df4396b297f4 100644
--- a/arch/x86/crypto/Makefile
+++ b/arch/x86/crypto/Makefile
@@ -92,7 +92,11 @@ nhpoly1305-avx2-y := nh-avx2-x86_64.o nhpoly1305-avx2-glue.o
 
 obj-$(CONFIG_CRYPTO_CURVE25519_X86) += curve25519-x86_64.o
 
+ifndef PERL_PATH
+  PERL_PATH = /usr/bin/perl
+endif
+
 quiet_cmd_perlasm = PERLASM $@
-      cmd_perlasm = $(PERL) $< > $@
+      cmd_perlasm = $(PERL_PATH) $< > $@
 $(obj)/%.S: $(src)/%.pl FORCE
 	$(call if_changed,perlasm)
-- 
2.17.1

